src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_StringConcat.cs

```diff
@@ -346,6 +406,156 @@ private BoundExpression RewriteStringConcatenationManyExprs(SyntaxNode syntax, I
             return BoundCall.Synthesized(syntax, receiverOpt: null, initialBindingReceiverIsSubjectToCloning: ThreeState.Unknown, method, array);
         }
 
+        private bool TryRewriteStringConcatenationWithSpanBasedConcat(SyntaxNode syntax, ArrayBuilder<BoundExpression> args, [NotNullWhen(true)] out BoundExpression? result)
+        {
+            var preparedArgs = ArrayBuilder<BoundExpression>.GetInstance(capacity: args.Count);
+
+            var needsSpanRefParamConstructor = false;
+            var needsImplicitConversionFromStringToSpan = false;
+
+            NamedTypeSymbol? charType = null;
+
+            foreach (var arg in args)
+            {
+                Debug.Assert(arg.HasAnyErrors || arg.Type?.IsStringType() == true);
+
+                if (arg is BoundCall { ReceiverOpt: { Type: NamedTypeSymbol { SpecialType: SpecialType.System_Char } receiverCharType } receiver } potentialToStringCall &&
+                    (object)potentialToStringCall.Method != _compilation.GetSpecialTypeMember(SpecialMember.System_Object__ToString) &&
```

Comment:
>(object)potentialToStringCall.Method != _compilation.GetSpecialTypeMember(SpecialMember.System_Object__ToString) && [](http://example.com/codeflow?start=20&length=115)

Sorry, I didn't mean to suggest that this case should get special treatment. Quite the opposite. We really do not care about unconventional behavior of special methods. In many different places we assume that `char.Tostring()` has specific behavior, regardless of whether the method is overridden by the `char` type or not. My questions were motivated by the desire to ensure that the treatment is uniform.

Reply:
So you would expect us to generate span-based concat regardless of the fact that `char.Tostring()` can be missing?